app/unit_test_bookmarks.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import app.prefs
+import app.text_buffer
+import app.window
+import mock
+import unittest
+
+class BookmarkTestCases(unittest.TestCase):
+  def setUp(self):
+    self.fakeHost = mock.Mock()
+    self.textBuffer = app.text_buffer.TextBuffer()
+    self.textBuffer.lines = 50
+    self.lineNumbers = app.window.LineNumbers(self.fakeHost)
```

Comment:
While mock is handy, it looks like this use can be replaced with a stub class. I'd like to avoid adding a dependency if it can be avoided.

Where there are dependencies, they are put in the third_party directory. That was done for pyperclip since pasteboard support is fairly fundamental (I haven't found a way around it). WDYT about not using mock here?

Reply:
Yes, I agree that these could be using a stub class instead. I just used them because I ended up having to import the mock library anyway, so it would look nicer to use a mock. One of the issues I was trying to solve was that `getBookmarkColor` references **app.prefs.startup**. For my tests, I wanted to set this value in order to test for the 8 and 256 colored terminals. However, I was thinking that if I were to change the dictionary directly and the tests were run in parallel (not really sure how they are run in github, or maybe elsewhere in the future), then this change could affect some other tests.

Maybe I was overthinking it and so I used a mock, but I also thought that since its for only used for running tests and it wouldn't affect the performance of the program, it would be okay. I could switch these mocks to be stub classes anyway in order to make the individual tests run faster, but I still think I need the `mock` library as I haven't found another way to stub out a dictionary like app.prefs.startup, besides mock.patch.

Reply:
Okay, I can try that. I thought that this wouldn't cause any trouble, since we could add this library in setup.py or something, which would end up minimizing any interaction the user has with installing any external dependencies. Perhaps even a docker image for developers to make it even easier. I'll try this exclusion method for now.

Reply:
Updated.